api/ws/rpc/connect.js

```diff
@@ -125,10 +125,9 @@ const connectSteps = {
 			(peerExtendedWithRPC, localHandler, rpcProcedureName) => {
 				peerExtendedWithRPC.rpc[rpcProcedureName] = (data, rpcCallback) => {
 					// Provide default parameters if called with non standard parameter, callback
+					const auxRpcCallback = typeof data === 'function' ? data : () => {};
 					rpcCallback =
-						typeof rpcCallback === 'function'
-							? rpcCallback
-							: typeof data === 'function' ? data : () => {};
+						typeof rpcCallback === 'function' ? rpcCallback : auxRpcCallback;
 					data = data && typeof data !== 'function' ? data : {};
```

Comment:
I think we can simplify the logic between lines 128-131 by:
```suggestion
					if (typeof data === 'function') {
						rpcCallback = data;
						data = {};
					}
```

I'm not even sure if we should set a default value like `() => {}` for rpcCallback. I think if not a valid callback provided we should throw an error since we are using setImmediate to return the result. But if it's necessary we can add the following condition on top of the above condition:

```suggestion
					if (typeof rpcCallback !== 'function') {
						rpcCallback = () => {};
					}
```

Reply:
I can understand if this can be considered as not scope of this issue. But since PR is relatively small maybe we can refactor these.